### PR TITLE
OSSKU supports Windows options

### PR DESCRIPTION
**Pull request by ShiqianTao**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [X] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
It has been released on 2022-04-03.
  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
As soon as possible, depends on the owner.
  6. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [X] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ShiqianTao** [comment]
Could you help to review this PR? Thank you.
_It seems that all PR failed on the part (SDK azure-sdk-for-net), and this PR is the same._
@AbelHu @FumingZhang @moreOver0

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json [hunk 1]
```diff
@@ -4868,13 +4868,15 @@
...
-        "CBLMariner"
+        "CBLMariner",
+        "Windows2019",
+        "Windows2022"
       ],
       "x-ms-enum": {
         "name": "OSSKU",
         "modelAsString": true
       },
-      "description": "Specifies an OS SKU. This value must not be specified if OSType is Windows."
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json:4877]: LGTM

So rp could handle `ossku = windows2019, ostype = linux` or other invalid combinations properly?

**Review by FumingZhang — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json (hunk 1 above)
Comment (ShiqianTao) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json:4877]: Yes. AKS-RP validates the OSSKU value for Linux and Windows.

**Review by AbelHu — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json (hunk 1 above)
Comment (matthchr) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json:4877]: Is there a default if one is not specified? For example `osType = Windows`, `osSKU` not set? If so we should document what that is (for both Windows/Linux)

**Review by matthchr — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json (hunk 1 above)
Comment (ShiqianTao) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json:4877]: Good guidance. Updated.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json [hunk 2]
```diff
@@ -4868,13 +4871,15 @@
...
+      "description": "Ubuntu or CBLMariner for Linux node pool. The default is Ubuntu if OSType=Linux. Windows2019 or Windows2022 for Windows node pool. The default is Windows2019 if OSType=Windows."
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json:4882]: minor: consider changing language slightly to clarify:
"Specifies the OS SKU used by the agent pool. If not specified, the default is Ubuntu if OSType=Linux or Windows2019 if OSType=Windows"

Comment (ShiqianTao) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json]: Thanks for your guidance. Updated the description.

Comment (AbelHu) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json]: @matthchr @ShiqianTao We may change the default OSSku from Windows2019 to Windows2022 when we decide to deprecate WS2019

Comment (AbelHu) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json]: should we update the description for Windows to `the default is Windows2019 if OSType=Windows and will be changed to Windows2022 after Windows2019 is deprecated`

Comment (ShiqianTao) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json]: Yes, it's more clear. Updated the description. I split it into two sentences.

**Review by AbelHu — APPROVED** [review]

**Review by palma21 — APPROVED** [review]

**Comment by ShiqianTao** [comment]
@ArcturusZhang Could you help to merge this PR? Thank you.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**